src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp

```diff
@@ -730,7 +730,9 @@ std::vector<cldnn::event::ptr> SyncInferRequest::prepare_input(const std::string
     if (!m_graph->get_config().get_property(ov::intel_gpu::allow_new_shape_infer)) {
         auto new_layout = memory->get_layout();
         new_layout.set_partial_shape(m_graph->get_input_layouts().at(name).get_shape());
-        memory = engine.reinterpret_buffer(*memory, new_layout);
+        memory = engine.reinterpret_buffer(*memory, cldnn::layout{ov::PartialShape{memory->get_layout().get_shape()},
```

Comment:
Memory is obtained from device tensor here, so it is supposed to have a valid data type. Please try to troubleshoot original data types mismatch